### PR TITLE
feat(business glossary): Add support to add & remove glossary terms from the UI

### PR DESCRIPTION
**Pull request by gabe-lyons**
Other changes included:

- migrate adding & removing tag updates to new addTag and removeTag mutations
- Deleting some legacy dataset files that are no longer used and referenced old tag methods
- Extending autocomplete to support autocompleting on a few entity types. This ended up not being needed for the PR since we have separate Add Tag and Add Term modals. However, I foresee this change being valuable down the line.

image

image

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/AddTermResolver.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import graphql.schema.DataFetchingEnvironment;
+import java.util.concurrent.CompletableFuture;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.linkedin.datahub.graphql.resolvers.ResolverUtils.*;
+
+
+public class AddTermResolver implements DataFetcher<CompletableFuture<Boolean>> {
+  private static final Logger _logger = LoggerFactory.getLogger(MutableTypeResolver.class.getName());
```
Comment (dexter-mh-lee) [inline datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/AddTermResolver.java:19]: If you add the @Slf4j annotation, you can just use log without having to add this variable

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/AddTermResolver.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+  private EntityService _entityService;
+
+  public AddTermResolver(EntityService entityService) {
```
Comment (dexter-mh-lee) [inline datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/AddTermResolver.java:23]: "@RequiredArgsConstructor" annotation on the class will automatically add  this

Comment (dexter-mh-lee) [reply datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/AddTermResolver.java:23]: Same for all the other resolvers

Comment (gabe-lyons) [reply datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/AddTermResolver.java]: addressed!

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/AddTermResolver.java (hunk 1 above)
Comment (gabe-lyons) [reply datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/AddTermResolver.java]: addressed!

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/AddTagResolver.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+    return CompletableFuture.supplyAsync(() -> {
+      try {
+
+        if (!tagUrn.getEntityType().equals("tag")) {
+          log.error(String.format("Failed to add %s. It is not a tag urn.", tagUrn.toString()));
+          return false;
+        }
+
+        log.info(String.format("Adding Tag. input: %s", input));
```
Comment (dexter-mh-lee) [inline datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/AddTagResolver.java:42]: nit: log.info("Adding Tag. input: {}", input)

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/LabelUtils.java [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+      PoliciesConfig.EDIT_ENTITY_PRIVILEGE.getType()
+  ));
+
+  private LabelUtils() { }
+
+  public static final String GLOSSARY_TERM_ASPECT_NAME = "glossaryTerms";
+  public static final String SCHEMA_ASPECT_NAME = "editableSchemaMetadata";
+  public static final String TAGS_ASPECT_NAME = "globalTags";
+
+  private static final Logger _logger = LoggerFactory.getLogger(LabelUtils.class.getName());
```
Comment (dexter-mh-lee) [inline datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/LabelUtils.java:44]: @Slf4j here as well

**Review by dexter-mh-lee — APPROVED** [review]
Overall looks good! Minor comments

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/LabelUtils.java (hunk 4 above)
Comment (gabe-lyons) [reply datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/LabelUtils.java]: Addressed!

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/AddTagResolver.java (hunk 3 above)
Comment (gabe-lyons) [reply datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/AddTagResolver.java]: done!

**Review by shirshanka — APPROVED** [review]
LGTM!

**Comment by ellard355** [comment]
Hi! Super cool changes that you're doing here. Could you give me some indication as to when this will be available in the Datahub software?

**Pull request merged**